packages/svelte/src/compiler/compile/render_dom/wrappers/InlineComponent/index.js

```diff
@@ -360,9 +360,15 @@ export default class InlineComponentWrapper extends Wrapper {
 					}
 				`;
 			}
+			// `skip_binding` is set by runtime/internal `bind()` function only at first call
+			// this prevents child -> parent backflow that triggers unnecessary $$.update (#4265)
+			// the flag is used to detech reactive mutation to object in `child.$$.update`
+			// ignore this flag when parent_value !== child_value
 			const body = b`
-				function ${id}(${params}) {
-					${invalidate_binding}
+				function ${id}(#skip_binding, ${params}) {
+					if (!#skip_binding || ${lhs} !== #value) {
+						${invalidate_binding}
+					}
```

Comment:
`if (!#skip_binding || ${lhs} !== #value)` guard prevents the unnecessary backflow of case 2, also `${lhs} !== #value` check allows backflow for case 1 and 4.1. 

Case 2: Child does not touch passed-in prop value, regardless it's primitive or object, backflow is unnecessary
Case 1 and 4.1: Child value is referentially different from Parent value